### DISCUSSION TITLE
Constrain dimensions of child group inside of a resizable group

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#694] [Q&A] [answered]
Does anyone know if there's a way to constrain the dimension of a child group that's inside of a resizable group (i.e, like a resizable window)? I would assume that calling ```make_resizable(false)``` on the child group would do this, but as you can see in my gif the child group is still resizing. I know I could use ```draw()``` to  force the child to keep certain dimensions, but this is a little cumbersome. 
```lang-rust
//fltk = "1.0.14"
use fltk::{app, button::Button, prelude::*, window::Window, enums::*, group::Group};
fn main() {
    let app = app::App::default();
    let mut wind = Window::default().with_size(400, 300);
    wind.make_resizable(true);
    let mut group = Group::new(0,0,200,200,"");
    group.make_resizable(false);
    let mut button = Button::new(0,0,100,100,"test");
    button.set_color(Color::Blue);
    group.set_frame(FrameType::FlatBox);
    group.set_color(Color::Red);
    group.end();
    wind.end();
    wind.show();
    app.run().unwrap();
}
``` 
Child group (red) resizes even though ```resizable()``` is ```false```:
![resize_issue](https://user-images.githubusercontent.com/38958118/119578099-e5e39a80-bd89-11eb-98bd-228edde7dcaf.gif)

**Comment by MoAlyousef** [comment, marked as answer]
You would need to add another hidden group:
```rust
use fltk::{app, button::Button, prelude::*, window::Window, enums::*, group::Group};
fn main() {
    let app = app::App::default();
    let mut wind = Window::default().with_size(400, 300);
    wind.make_resizable(true);
    let mut group1 = Group::new(0, 0, 200, 200, "");
    group1.make_resizable(false);
    let mut group2 = Group::new(0,0,200,200,"");
    let mut button = Button::new(0,0,100,100,"test");
    button.set_color(Color::Blue);
    group2.set_frame(FrameType::FlatBox);
    group2.set_color(Color::Red);
    group2.end();
    group1.end();
    wind.end();
    wind.show();
    app.run().unwrap();
}
```

**Reply by wyhinton** [reply]
Thanks for your answer! So if a parent is resizable, all children are forced to be resizable?

**Reply by MoAlyousef** [reply]
No problem. It’s a but complicated. A resizable group/window widget makes children resizable. If you want to specify widgets, you’ll have to pass them individually to `GroupExt::resizable` or override things by adding another group and making that group not resizable.
